### ISSUE TITLE
ClearML integration removes the best checkpoint after uploading to the server

### ISSUE THREAD

**Issue by kecsap** [ultralytics/yolov5#9251]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Integrations

### Bug

YOLOv5 Git is up-to-date.
After configuring a local ClearML server, everything works fine, but the integration removes the uploaded best checkpoint file (best.pt) from runs/train/expXX/weights/ after the training is finished and it is uploaded to the server.

### Environment

Ubuntu 20.04

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@thepycoder is the logging code deleting local checkpoints on upload?

**Comment by Denizzje** [comment]
Oh... so this might explain why I have not seen any best.pt at all having trained with various model sizes from YoloV5 v6.2. I geniunely thought this was a change from 6.1 to 6.2 (we use the fixed releases for training due to some modifications in training code). But I also enabled clearML, but to their cloud service for now instead of local. None of my trained models had a best.pt in the weights folder after training. Looking at the ClearML artificats of each run, I do see them being there now.

**Comment by glenn-jocher** [comment]
@Denizzje no, the only reason best.pt would be missing is if you trained with --nosave or --noval.

**Comment by kecsap** [comment]
ClearML installed:

```bash
$ python3 train.py --img 640 --batch 124 --epochs 3 --data XXX.yaml --weights yolov5s.pt --freeze 10
train: weights=yolov5s.pt, cfg=, data=XXX.yaml, hyp=data/hyps/hyp.scratch-low.yaml, epochs=3, batch_size=124, imgsz=640, rect=False, resume=False, nosave=False, noval=False, noautoanchor=False, noplots=False, evolve=None, bucket=, cache=None, image_weights=False, device=, multi_scale=False, single_cls=False, optimizer=SGD, sync_bn=False, workers=8, project=runs/train, name=exp, exist_ok=False, quad=False, cos_lr=False, label_smoothing=0.0, patience=100, freeze=[10], save_period=-1, seed=0, local_rank=-1, entity=None, upload_dataset=False, bbox_interval=-1, artifact_alias=latest
...
3 epochs completed in 0.002 hours.
Optimizer stripped from runs/train/exp92/weights/last.pt, 14.8MB
Optimizer stripped from runs/train/exp92/weights/best.pt, 14.8MB
...
Validating runs/train/exp92/weights/best.pt...
...
Results saved to runs/train/exp92
2022-09-02 11:49:55,696 - clearml.Task - INFO - Waiting to finish uploads
2022-09-02 11:49:55,801 - clearml.Task - INFO - Completed model upload to http://X.X.X.X:8081/YOLOv5/training.XXX/models/best.pt
2022-09-02 11:50:01,036 - clearml.Task - INFO - Finished uploading

$ ls runs/train/exp92/weights/best.pt
ls: cannot access 'runs/train/exp92/weights/best.pt': No such file or directory
```

Repearing the same after `pip3 uninstall clearml`:

```bash
$ python3 train.py --img 640 --batch 124 --epochs 3 --data XXX.yaml --weights yolov5s.pt --freeze 10
train: weights=yolov5s.pt, cfg=, data=XXX.yaml, hyp=data/hyps/hyp.scratch-low.yaml, epochs=3, batch_size=124, imgsz=640, rect=False, resume=False, nosave=False, noval=False, noautoanchor=False, noplots=False, evolve=None, bucket=, cache=None, image_weights=False, device=, multi_scale=False, single_cls=False, optimizer=SGD, sync_bn=False, workers=8, project=runs/train, name=exp, exist_ok=False, quad=False, cos_lr=False, label_smoothing=0.0, patience=100, freeze=[10], save_period=-1, seed=0, local_rank=-1, entity=None, upload_dataset=False, bbox_interval=-1, artifact_alias=latest
...
3 epochs completed in 0.002 hours.
Optimizer stripped from runs/train/exp93/weights/last.pt, 14.8MB
Optimizer stripped from runs/train/exp93/weights/best.pt, 14.8MB
...
Validating runs/train/exp93/weights/best.pt...
...
Results saved to runs/train/exp93

$ ls runs/train/exp93/weights/best.pt
runs/train/exp93/weights/best.pt
```

**Comment by glenn-jocher** [comment]
@kecsap thanks for the example! @thepycoder per the example in https://github.com/ultralytics/yolov5/issues/9251#issuecomment-1235308049 it appears the ClearML integration is moving or deleting best.pt after training completes. 

Can you try to reproduce and investigate a fix? Thanks!

**Comment by thepycoder** [comment]
Hey @kecsap,

Thank you so much for the reproducible example. That is very weird indeed. I'm looking into it!

For others that might have the issue: you can still retrieve your best model by going to the experiment webui and downloading it there or by using the sdk

```
task = Task.get_task(task_id='previously_executed_task') my_local_copy_of_the_previously_uploaded_artifact = task.artifacts["best"].get_local_copy()
```

Not at a computer right now, but will fix asap!

**Comment by thepycoder** [comment]
Hey @glenn-jocher, @kecsap 

I opened a PR with a fix. Sorry for the inconvenience!

**Comment by yuyang3478** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by glenn-jocher** [comment]
@kecsap good news 😃! Your original issue may now be fixed ✅ in PR #9265. To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
